src/main/webapp/creatives/maps/simid_maps.js

```diff
@@ -1,8 +1,44 @@
 /**
  * A sample SIMID ad that shows a map of nearby locations
  */
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+}
+const DEFAULT_BUTTON_LABEL = "Location";
+
 class SimidMapCreative extends BaseSimidCreative {
   constructor() {
     super();
   }
+
+  /** @override */
+  onStart(eventData){
+    super.onStart(eventData);
+    const adParams = this.creativeData.adParameters;
+    var buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL]; 
+    if (buttonLabel == undefined) {
+      buttonLabel = DEFAULT_BUTTON_LABEL;
+    }
+    else {
+      buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL];
+    }
+    var searchQuery = JSON.parse(adParams)[AdParamKeys.SEARCH_QUERY];
```

Comment:
Shouldn't these variables be declared as const?

Reply:
nope they will be probably be used in other functions

Reply:
use let instead

Reply:
If I remember correctly Dan told us to avoid using var in JavaScript, the style guide seems to support this https://google.github.io/styleguide/jsguide.html#features-use-const-and-let. If they're going to be used in other functions it would be clearer to declare as const (or let if they ever need to be modified) and pass them into the function as parameters.

Reply:
i will change to let